extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -371,29 +372,39 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
 
 	protected function sendEmails( array $watchers, WallNotificationEntity $notification ) {
 		$text = strip_tags( $notification->data_noncached->msg_text, '<p><br>' );
-		$text = substr( $text, 0, 3000 ) . ( strlen( $text ) > 3000 ? '...' : '' );
-
-		$textNoHtml = preg_replace( '#<br\s*/?>#i', "\n", $text );
-		$textNoHtml = trim( preg_replace( '#</?p\s*/?>#i', "\n", $textNoHtml ) );
-		$textNoHtml = substr( $textNoHtml, 0, 3000) . ( strlen( $textNoHtml ) > 3000 ? '...' : '' );
+		$text = mb_substr( $text, 0, self::MAX_ABSTRACT_LENGTH, 'UTF-8' )
```

Comment:
Not crucial, but all this text preparation seems like a good candidate for it's own method.